### PR TITLE
Update changelog and documentation

### PR DESCRIPTION
**Pull request by CezaryGapinski**
I'm trying to update the documentation and I found some issues during working on ity, so maybe anybody else could help me.

- I have noticed incorrectly generated html output for Sphinx 2.0. For me everything works for version 1.8.5, but when version is higher then I have noticed incorrect formatting in the `guide.html` (ex. listing with example.cpp is in one line). Anybody else also have this problem or it is something wrong with my configuration?
- How `PYTHON` variable in doc/Makefile should be set by default? Now is set to the `python`, but in the gcovr Makefile is set to `python3`.
- Is description about testing from CONTRIBUTING.rst is up-to-date?

After #374 we have new options to generate/update/skip clean references, but 
`python3 -m pytest -k xml`
mentioned in the documentation causing an error:

```
============================================================================================= ERRORS =============================================================================================
___________________________________________________________________________ ERROR collecting gcovr/tests/test_gcovr.py ___________________________________________________________________________
../../.local/lib/python3.8/site-packages/_pytest/config/__init__.py:1155: in getoption
    val = getattr(self.option, name)
E   AttributeError: 'Namespace' object has no attribute 'skip_clean'
During handling of the above exception, another exception occurred:
gcovr/tests/test_gcovr.py:130: in pytest_generate_tests
    skip_clean = metafunc.config.getoption("skip_clean")
../../.local/lib/python3.8/site-packages/_pytest/config/__init__.py:1166: in getoption
    raise ValueError("no option named {!r}".format(name))
E   ValueError: no option named 'skip_clean'
```

It will be great to explain how to use `pytest` after those changes. @spacetown, maybe you can help to explain usage of `pytest` in the CONTRIBUTING.rst?

### CODE REVIEW

**Comment by Spacetown** [comment]
In CONTRIBUTING.rst the command `make test` is mentioned and a reference to the test suite. In the test suite section the documentation for the make variable `TEST_OPTS` is missing. The command used for the test is:
```
GCOVR_TEST_SUITE=1 CC=$(CC) CXX=$(CXX) GCOV=$(GCOV) $(PYTHON) -m pytest -v --doctest-modules $(TEST_OPTS) -- gcovr doc/examples
```
The variable `TEST_OPTS`can be used in this ways:
- `make test TEST_OPTS="-k 'xml'"`
- `make test TEST_OPTS="-k 'html' --update_reference"`
- ...

I would be grateful if you mention the variable in the documentation. I am now on holiday for 10 days and have no access.

Your `AttributeError` about the missing `skip_clean` can be from the otpions of pytest. It seems, that the file gcovr/tests/conftest.py is ignored in your call.

**Comment by CezaryGapinski** [comment]
Ok, I've added mention about run tests only by `make test` with options passed via `TEST_OPTS` variable.

@Spacetown Have a nice vacation!

**Comment by Spacetown** [comment]
@CezaryGapinski Thanks.
The variable isn't new. I've only updated the Docker command to inject the variable into the container a few commits ago and introduced the options for the test. Before this the "generate reference" was a global constant before.

**Review by Spacetown — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -39,9 +41,10 @@ Documentation:
...
 
  - Support multiple output patterns in integration tests (:issue:`383`)
- - Extend tests to use an unified diff in the assert. Add test options `--generate-reference`,
+ - Extend tests to use an unified diff in the assert. Add test options `--generate_reference`,
    `--update_reference` and `--skip_clean`. (:issue:`374`)
  - Set make variable TEST_OPTS as environment variable inside docker. (:issue: `372`)
+ - Enable :option:`--filter` and :option:`--exclude` for :ref:`Combining tracefiles <combining_tracefiles>`. (:issue:`373`)
  - Extend test framework for CI (:issue:`392`):
 
    - Extend test option `--update_reference` for all formats.
```
[the remark this answers is not in the document]
Comment (Spacetown) [reply CHANGELOG.rst:50]: Line 50.

**Review by Spacetown — APPROVED** [review]

**Pull request merged**